framework/src/main/java/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesHelper.java

```diff
@@ -86,6 +87,24 @@
      */
     private final Set<String> modifiedScenes = new HashSet<>();
 
+    /**
+     * AScene doesn't carry around the basetypes of annotated types, which precludes outputting
+     * correctly formatted stub files. #updateTypeElementFromATM intercepts updates to
+     * ATypeElement_s that add annotations, and populates this map with their corresponding
+     * basetypes. Then, these basetypes are used when outputting stub files.
+     *
+     * <p>This is super hacky, and could be avoided if we didn't go through AScene at all.
+     * Unfortunately, all the interesting WPI code uses them, so removing that dependency would be
+     * hard.
+     *
+     * <p>The ATypeElement_s themselves aren't useful as keys, because their hashes change when
+     * annotations are added to them. Instead, we use their description as the hash key. This is
+     * super sketchy (there is no guarantee that two different ATypeElement_s won't have the same
+     * description!), but it's the best I can come up with at the moment.
+     */
+    private final Map<String, TypeMirror> basetypes = new HashMap<>();
```

Comment:
Again, start the comment with what this field is, then add explanation.  Also, `super hacky` isn't a help comment as it does not explain what the `hack` part it.  I think the problem is that two ATypeElement_s may have the same description, but different TypeMirrors, but I'm not sure.